sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -47,11 +47,16 @@ class IncrementalExecution(
       sparkSession.sparkContext,
       sparkSession.sessionState.conf,
       sparkSession.sessionState.experimentalMethods) {
-    override def extraPlanningStrategies: Seq[Strategy] =
-      StatefulAggregationStrategy ::
-      FlatMapGroupsWithStateStrategy ::
-      StreamingRelationStrategy ::
-      StreamingDeduplicationStrategy :: Nil
+    override def strategies: Seq[Strategy] =
+      sparkSession.sessionState.planner.strategies
```

Comment:
I think this should be `extraPlanningStrategies  ++ sparkSession.sessionState.planner.strategies`, and you also need to remove `sparkSession.sessionState.planner.extraPlanningStrategies` from `extraPlanningStrategies`.